### PR TITLE
Collision-with-Diagonals

### PR DESCRIPTION
**Pull request by Pathfinder-Liam**
Pull request created for help with failure to access self.list_of_corners attribute created on wall class

### CODE REVIEW

File: src/arena/wall.py [hunk 1]
```diff
@@ -29,5 +31,12 @@ def init_for_drawing(self):
             int(self._size[0]), int(self._size[1]), color.BLACK
         )
         self._sprite.owner = self
+        self.upper_left: tuple = (self._sprite.left, self._sprite.top)
+        self.upper_right: tuple = (self._sprite.right, self._sprite.top)
+        self.bottom_left: tuple = (self._sprite.left, self._sprite.bottom)
+        self.bottom_right: tuple = (self._sprite.right, self._sprite.bottom)
+        self.list_of_corners: list[tuple](
+            self.upper_left, self.upper_right, self.bottom_right, self.bottom_left
+        )
```
[the remark this answers is not in the document]
Comment (Pathfinder-Liam) [reply src/arena/wall.py:40]: Oh no, here I was thinking I found a good way to find the vertices. thanks for catching this
https://api.arcade.academy/en/latest/api/physics_engines.html is looking better every day

File: src/collision.py [hunk 2]
```diff
@@ -24,4 +28,56 @@ def projectile_hits_player(delta_time, sprite_lists: SpriteLists):
...
-            projectile_sprite.owner.on_collision_with_player(delta_time, players_touching_projectile)
+            projectile_sprite.owner.on_collision_with_player(
+                delta_time, players_touching_projectile
+            )
+
+
+def player_hits_wall(sprite_lists: SpriteLists):
+    for player_sprite in sprite_lists.players:
+        player_sprite: LinkedSprite[Player]
+        walls_probably_touching: LinkedSprite[Wall]
```
[the remark this answers is not in the document]
Comment (Pathfinder-Liam) [reply src/collision.py:39]: fixed this

File: src/collision.py [hunk 3]
```diff
@@ -24,4 +28,56 @@ def projectile_hits_player(delta_time, sprite_lists: SpriteLists):
...
+        walls_touching_players = arcade.check_for_collision_with_list(
```
[the remark this answers is not in the document]
Comment (Pathfinder-Liam) [reply src/collision.py:40]: changed this

File: src/arena/wall.py (hunk 1 above)
Comment (Thorakii) [inline src/arena/wall.py:40]: I pinpointed what's throwing the error when the car drives into the wall. missing equals sign.
self.list_of_corners: list[tuple] = (
            self.upper_left, self.upper_right, self.bottom_right, self.bottom_left
        )
Once fixed, the car still drives through the diagonal wall, but doesn't throw an error

File: src/player.py [hunk 4]
```diff
@@ -44,6 +43,17 @@ def __init__(
...
+        self.upper_left: tuple = (self.sprite.left, self.sprite.top)
+        self.upper_right: tuple = (self.sprite.right, self.sprite.top)
+        self.bottom_left: tuple = (self.sprite.left, self.sprite.bottom)
+        self.bottom_right: tuple = (self.sprite.right, self.sprite.bottom)
+        self.list_of_corners: list[tuple] = (
+            self.upper_right,
+            self.upper_right,
+            self.bottom_right,
+            self.bottom_left,
+        )
```
[the remark this answers is not in the document]
Comment (Pathfinder-Liam) [reply src/player.py:55]: That did the trick, unfortunately I don't actually have the real vertices of the shapes so it doesn't work (probably among other things) 
Thanks for looking Andrew!

**Pull request closed without merging**